### PR TITLE
#125 Add support for deserializing on to an IndexedSeq

### PR DESCRIPTION
**Pull request by kennymacleod**
Changes `SeqDeserializer` to support deserializing on to properties of type `IndexedSeq`.

### CODE REVIEW

**Comment by kennymacleod** [comment]
Tests are failing, but seem to be failing on master also.

I added a new test to cover the change.

File: src/main/java/io/vavr/jackson/datatype/deserialize/SeqDeserializer.java [hunk 1]
```diff
@@ -46,6 +47,9 @@
         if (Array.class.isAssignableFrom(javaType.getRawClass())) {
             return Array.ofAll(result);
         }
+        if (IndexedSeq.class.isAssignableFrom(javaType.getRawClass())) {
```
Comment (ruslansennov) [inline src/main/java/io/vavr/jackson/datatype/deserialize/SeqDeserializer.java]: Hi @kennymacleod 

> Tests are failing, but seem to be failing on master also.

Just move this one to the end of the method (right before "default serialization"), and I believe that all tests will pass. The Vavr's `Vector` is `IndexedSeq` also.

Comment (kennymacleod) [reply src/main/java/io/vavr/jackson/datatype/deserialize/SeqDeserializer.java]: Good call. Done.

**Comment by ruslansennov** [comment]
thank you

**Pull request merged**